### ISSUE TITLE
[Diagnostic ER] Missing unnecessary parentheses warning

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust#96606]
rustc could give a unnecessary parentheses warning here:

```rust
fn main() {
    let arr = [0; 10];
    let i = 2;
    let y = arr[(5 - i)];
    dbg!(y);
}
```
